platforms/core-configuration/configuration-cache/build.gradle.kts

```diff
@@ -123,6 +123,14 @@ dependencies {
     crossVersionTestDistributionRuntimeOnly(projects.distributionsCore)
 }
 
+jvmCompile {
+    compilations {
+        named("main") {
+            targetJvmVersion = 17
```

Comment:
I have plans to follow-up and properly use the `gradleModule`, `usedInWorkers`, etc flags, but would like to get this in to lock-in the changes to testing